src/rinoh/frontend/sphinx/__init__.py

```diff
@@ -195,6 +195,10 @@ def assemble_doctree(self, indexfile, toctree_only):
         largetree = inline_all_toctrees(self, self._docnames, indexfile,
                                         new_tree, darkgreen, [indexfile])
         largetree['docname'] = indexfile
+        largetree.citations.extend(tree.citations)
+        largetree.citation_refs.update(tree.citation_refs)
+        largetree.ids.update(tree.ids)
+        largetree.refnames.update(tree.refnames)
```

Comment:
These seem to be unnecessary? The LaTeX builder doesn't seem to do this either.

Reply:
Sorry, this was left over from when I was trying to figure out where the problem was.

`assemble_doctree()` makes a copy of the doctree (minus the references metadata) before modifying it. This is one of the differences between sphinx and docutils.

What is the reason that `assemble_doctree()` makes a copy rather than using tranformations on the tree itself?

Reply:
> What is the reason that `assemble_doctree()` makes a copy rather than using tranformations on the tree itself?

I started basically copied how the LaTeX builder works and made adjustments as needed. The LaTeX builder also copies the doctree (actually, `inline_all_doctrees()` does). Not sure what the reason for that is. I was first thinking that a builder shouldn't modify the doctree because it may passed to other builders as well. But as far as I know, it's not possible to have Sphinx build a project with multiple builders in a single run...